france_connect/clients.py

```diff
@@ -0,0 +1,192 @@
+import secrets
+import urllib.parse
+from typing import List, Tuple
+
+import jwt
+import requests
+from jwt import ExpiredSignatureError, InvalidTokenError, PyJWKClient
+
+from .enums import ACRValues, Scopes
+from .exceptions import (
+    FranceConnectAccessTokenError,
+    FranceConnectConfigurationError,
+    FranceConnectExpiredTokenError,
+    FranceConnectHttpError,
+    FranceConnectInvalidTokenError,
+)
+
+
+class FranceConnect:
+    """
+    FranceConnect Client.
+
+    This class represents a client for interacting with the FranceConnect API.
+    It provides methods for authentication, token exchange, and user information retrieval.
+
+
+    Parameters
+    ----------
+    client_id (str): The client ID for the FranceConnect application.
+    client_secret (str): The client secret for the FranceConnect application.
+    scopes (List[Scopes]): The list of scopes to request during authentication.
+    post_login_callback_url (str): The URL to redirect to after successful login.
+    post_logout_callback_url (str): The URL to redirect to after logout.
+    fc_base_url (str): The base URL of the FranceConnect API.
+    fc_authorization_url (str, optional): The authorization URL of the FranceConnect API.
+    fc_token_url (str, optional): The token URL of the FranceConnect API.
+    fc_jwks_url (str, optional): The JWKS URL of the FranceConnect API.
+    fc_userinfo_url (str, optional): The userinfo URL of the FranceConnect API.
+    fc_logout_url (str, optional): The logout URL of the FranceConnect API.
+    """
```

Comment:
You should use numpy convention as set in `setup.cfg` (I'm surprised pydocstyle do not complain):

```
    """Client allowing interaction with the FranceConnect API

    Provides methods for authentication, token exchange, and user information
    retrieval.

    Attibutes
    ---------
    client_id: str
        The client ID for the FranceConnect application.
    client_secret: str
        The client secret for the FranceConnect application.
    scopes: List[Scopes]
        The list of scopes to request during authentication.
    post_login_callback_url: str
        The URL to redirect to after successful login.
    post_logout_callback_url: str
        The URL to redirect to after logout.
    fc_base_url: str
        The base URL of the FranceConnect API.
    fc_authorization_url: str, optional
        The authorization URL of the FranceConnect API.
    fc_token_url: str, optional
        The token URL of the FranceConnect API.
    fc_jwks_url: str, optional
        The JWKS URL of the FranceConnect API.
    fc_userinfo_url: str, optional
        The userinfo URL of the FranceConnect API.
    fc_logout_url: str, optional
        The logout URL of the FranceConnect API.
    """
```